### ISSUE TITLE
Web Clipper Default Interpreter per Template

### ISSUE THREAD

**Issue by graphicagenda** [obsidianmd/obsidian-clipper#242]
# Use case
I’ve updated to the newest Web Clipper version (0.10.1), and re-inserted my keys.
I was hoping that the next version of Web Clipper will allow us to set the default interpreter used within each template. That way, we could turn on the “Automatically Run” behavior, since the template had loaded the interpreter of choice.

For example, I have 2 Web clippings templates, each that works best with a different interpreter.

# Proposed solution

Add a new “default interpreter” option in a single Template’s setting.
This new setting would looks similar to the Vault chooser, it could have a “Last Used”, and then it could list all the available interpreters added from the Interpreter options page.

# Current workaround (optional)

1. Turning off “Automatically run” for the Interpreter behavior.
2. Choosing the interpreter for each template that has an interpreter.
3. Interpret

**Comment by kepano** [comment]
I am a bit afraid of adding such a feature because it would add complexity to the template form, and more possibilities for users to get confused when the model appearing in the popup is not the one they expect. Can you share more detail about why those templates/sites perform better with different models?

**Comment by fabioscarsi** [comment]
> # Use case
> I’ve updated to the newest Web Clipper version (0.10.1), and re-inserted my keys. I was hoping that the next version of Web Clipper will allow us to set the default interpreter used within each template. That way, we could turn on the “Automatically Run” behavior, since the template had loaded the interpreter of choice.
> 
> For example, I have 2 Web clippings templates, each that works best with a different interpreter.
> 
> # Proposed solution
> Add a new “default interpreter” option in a single Template’s setting. This new setting would looks similar to the Vault chooser, it could have a “Last Used”, and then it could list all the available interpreters added from the Interpreter options page.
> 
> # Current workaround (optional)
> 1. Turning off “Automatically run” for the Interpreter behavior.
> 2. Choosing the interpreter for each template that has an interpreter.
> 3. Interpret

I agree absolutely. In a template, the prompt can be straightforward to interpret, but in other templates, the interpreting job can be far more complex and, in reality, made of more than one prompt. In the first case, we can use a cheaper and faster LLM, but in the last, we must use a top and slower LLM. I have very different results depending on the LLM I use. For some templates, only a few LLMs can interpret the prompts contained in the template.
